File: Apple Capture/src/client/Controllers/RoundController.lua
local Knit = require(game:GetService('ReplicatedStorage').Knit);

local INTERMISSION_EVENT_NAME = 'Intermission';
local ROUND_EVENT_NAME = 'Round';
local COLLECTED_EVENT_NAME = 'Collected';
local WINNER_CHOSEN_EVENT_NAME = 'WinnerChosen';

local infoGUI = game:GetService('Players').LocalPlayer.PlayerGui:WaitForChild('InfoGUI');
local infoLabel = infoGUI:WaitForChild('InfoLabel');
local collectedApplesLabel = infoLabel:WaitForChild('CollectedApplesLabel');

local RoundController = Knit.CreateController({
    Name = 'RoundController'
});

function RoundController:KnitStart()
    Knit.GetService('RoundService').RoundInfo:Connect(function(event, ...)
        if event == INTERMISSION_EVENT_NAME then
            local secondsRemaining = ...;
            infoLabel.Text = event .. ': ' .. secondsRemaining;
        elseif event == ROUND_EVENT_NAME then
            local secondsRemaining = ...;
            infoLabel.Text = event .. ': ' .. secondsRemaining;
        elseif event == COLLECTED_EVENT_NAME then
            local collectedApples = ...;
            collectedApplesLabel.Text = collectedApples;
        elseif event == WINNER_CHOSEN_EVENT_NAME then
            local winner, mostCollectedApples = ...;
            if winner then
                infoLabel.Text = winner .. ' won the round with ' .. mostCollectedApples .. ((mostCollectedApples == 1) and ' apple!' or ' apples!');
            else
                infoLabel.Text = 'Nobody won the round!';
            end;
        end;
    end);
end;

function RoundController:KnitInit()

end;

return RoundController;

File: AppleCaptureTS/src/client/controllers/round-controller.ts
import { KnitClient as Knit } from "@rbxts/knit";
import { Players } from "@rbxts/services";

const INTERMISSION_EVENT_NAME = 'Intermission';
const ROUND_EVENT_NAME = 'Round';
const COLLECTED_EVENT_NAME = 'Collected';
const WINNER_CHOSEN_EVENT_NAME = 'WinnerChosen';

const infoGUI = (Players.LocalPlayer.FindFirstChildOfClass('PlayerGui') as PlayerGui).WaitForChild('InfoGUI') as ScreenGui;
const infoLabel = infoGUI.WaitForChild('InfoLabel') as TextLabel;
const collectedApplesLabel = infoLabel.WaitForChild('CollectedApplesLabel') as TextLabel;

declare global {
    interface KnitControllers {
        RoundController: typeof RoundController;
    }
}

const RoundController = Knit.CreateController({
    Name: 'RoundController',

    KnitStart() {
        Knit.GetService('RoundService').RoundInfo.Connect((eventName: string, data: number | string | undefined, mostCollectedApples?: number) => {
            if ((eventName === ROUND_EVENT_NAME) || (eventName === INTERMISSION_EVENT_NAME)) {
                const secondsRemaining = data as number;
                infoLabel.Text = `${eventName}: ${tostring(secondsRemaining)}`;
            } else if (eventName === COLLECTED_EVENT_NAME) {
                const collectedApples = data as number;
                collectedApplesLabel.Text = tostring(collectedApples);
            } else if (eventName === WINNER_CHOSEN_EVENT_NAME) {
                let winner = data as string;
                if (winner) {
                    infoLabel.Text = `${winner} won the round with ${mostCollectedApples} ${(mostCollectedApples === 1) ? 'apple!' : 'apples!'}`;
                } else {
                    infoLabel.Text = 'Nobody won the round!';
                }
            }
        });
    }
});